diff -u -r boost-1.89.0.orig/libs/cobalt/CMakeLists.txt boost-1.89.0/libs/cobalt/CMakeLists.txt
--- boost-1.89.0.orig/libs/cobalt/CMakeLists.txt        2025-08-14 09:26:07.000000000 +0800
+++ boost-1.89.0/libs/cobalt/CMakeLists.txt     2025-08-19 14:49:21.861899800 +0800
@@ -73,7 +73,7 @@
     endif()
 
     if(MINGW OR CYGWIN)
-        target_link_libraries(boost_cobalt PUBLIC ws2_32)
+        target_link_libraries(boost_cobalt PUBLIC ws2_32 mswsock bcrypt)
     endif()
 
     if (BOOST_COBALT_USE_BOOST_CONTAINER)
@@ -210,7 +210,7 @@
     endif()
 
     if(MINGW OR CYGWIN)
-        target_link_libraries(boost_cobalt PUBLIC ws2_32)
+        target_link_libraries(boost_cobalt PUBLIC ws2_32 mswsock bcrypt)
     endif()
 
     if(BUILD_SHARED_LIBS)
